52 password policy enforcement

sudheerg9 opened the pull request:
@tmichalski  - Please review the changes made as per your suggestions for PR on vaoyage-java-api

tmichalski commented on src/test/groovy/voyage/security/user/UserServiceSpec.groovy:
```diff
@@ -367,4 +380,57 @@ class UserServiceSpec extends Specification {
             userRepository.findOne(_) >> user
             user.isDeleted
     }
+    def 'update password - updating password with new password should compare passwords and saving user object'() {
```
Add a line break before the method signature.

tmichalski commented:
Looks good except that I don't see any tests that execute the password validation to verify that it is functional. In UserService the tests mock the PasswordValidator and test the UserService functionality. There probably should be a test (integration test?) that validates that the Config is being loaded properly and that passwords are being validated by Passy / PasswordValidator bean according to the configuration. 

Please add these tests and then this work will be complete.

sudheerg9 commented:
@tmichalski - I have updated the test cases by introducing a new Service Integration class. For now, I added test cases for only password validation. Let me know if we need to add test cases for all the existing functionality. so, I will take up the task separately and complete it.
Added custom messages as well in WeakPasswordException by considering the Error codes we will get from Passay library, please review that approach as well.
